spot_rl_experiments/spot_rl/envs/place_env.py

```diff
@@ -2,69 +2,249 @@
+
+                # Get direction vector from current ee position to goal ee position for EE orientation
+                dir_rpy_to_goal = get_RPY_from_vector(goal_ee_pos - curr_ee_pos)
+                print(dir_rpy_to_goal)
+
+                # Go to intermediate point
+                cmd_id = self.spot.move_gripper_to_point(
+                    intr_ee_pos, dir_rpy_to_goal, 1
+                )
+                self.spot.block_until_arm_arrives(cmd_id, timeout_sec=10)
+
+                # Direct the gripper to face downwards
+                dir_rpy_to_goal = [0.0, np.pi / 2, 0.0]
+
+                # Go to goal point
+                cmd_id = self.spot.move_gripper_to_point(
+                    goal_ee_pos, dir_rpy_to_goal, 1
+                )
+                self.spot.block_until_arm_arrives(cmd_id, timeout_sec=10)
+
+            success_list.append(
+                {
+                    # "place_object": "place_target",
+                    "time_taken": time.time() - start_time,
+                    "success": self.place_env.place_attempted or not self.use_policies,
+                    "place_target": self.get_place_target_in_base_frame(),
+                    "ee_pos": self.get_gripper_position_in_base_frame(),
+                }
+            )
+
+            # Open gripper to drop the object
+            self.spot.open_gripper()
+            # Add sleep as open_gripper() is a non-blocking call
+            time.sleep(1)
+
+        return success_list
+
+    # Should this be kept here or moved to spot.py?????
+    def get_gripper_position_in_base_frame(self):
+        """
+        Returns the gripper position in the base frame as (x,y,z)
+
+        Returns:
+            ee_pos: Gripper position as an np.array([x,y,z])
+        """
+        # Get end effector position (in base frame) .. (x,z,-y) -> (x,-y,z)
+        ee_pos = np.array(self.place_env.get_in_gripper_tf().translation)[
+            np.array([0, 2, 1])
+        ]
+        # Correct the sign on "y"
+        ee_pos[1] *= -1
+        return ee_pos
+
+    # Should this be kept here or moved to spot.py?????
+    def get_place_target_in_base_frame(self):
+        """
+        Returns the current place target in the base frame as (x,y,z)
+
+        Returns:
+            base_frame_place_target_corrected: Place target in base frame as an np.array([x,y,z])
+        """
+        # Get place target (in base frame) .. x,z,-y -> (x,-y,z)
+        base_frame_place_target_uncorrected = (
+            self.place_env.get_base_frame_place_target()
+        )
+        base_frame_place_target_corrected = np.array(
+            self.place_env.spot2habitat_translation(base_frame_place_target_uncorrected)
+        )[np.array([0, 2, 1])]
+
+        # Correct the sign on "y"
+        base_frame_place_target_corrected[1] *= -1
+        return base_frame_place_target_corrected
+
+    def shutdown(self, should_dock=False):
```

Comment:
is it possible to disentangle env and the shutdown procedure? maybe something like `shutdown(env)` in `spot.py`

Reply:
This will be a seperate PR

Reply:
This method has a dependence on other functions inside `base_env`.
So cannot move it to `spot.py`.

Moved it inside `base_env.py` 's `BaseEnv` class

Reply:
This is covered by issue - https://github.com/facebookresearch/spot-sim2real/issues/68

This is Solved by PR - https://github.com/facebookresearch/spot-sim2real/pull/66